internal/core/window.rs

```diff
@@ -685,15 +685,16 @@ impl WindowInner {
         event.modifiers = self.modifiers.get().into();
 
         let mut item = self.focus_item.borrow().clone().upgrade();
+
+        if item.as_ref().is_some_and(|i| !i.is_visible()) {
+            // Reset the focus... not great, but better than keeping it.
```

Comment:
In the future, we could introduce some kind of "EnsureVisible" event, and if the item accepts that (i.e. it moved itself into visible space), then we could continue.